src/Microsoft.Health.Fhir.SqlServer/Features/Schema/Sql/Tables/Resource.sql

```diff
@@ -1,52 +1,48 @@
+   ,IsHistory                   bit                     NOT NULL CONSTRAINT DF_ResourceCurrent_IsHistory DEFAULT 0, CONSTRAINT CH_ResourceCurrent_IsHistory CHECK (IsHistory = 0)
+   ,IsDeleted                   bit                     NOT NULL
+   ,RequestMethod               varchar(10)             NULL
+   ,RawResource                 varbinary(max)          NOT NULL CONSTRAINT CH_ResourceCurrent_RawResource_Length CHECK (RawResource > 0x0)
+   ,IsRawResourceMetaSet        bit                     NOT NULL CONSTRAINT DF_ResourceCurrent_IsRawResourceMetaSet DEFAULT 0
+   ,SearchParamHash             varchar(64)             NULL
+   ,TransactionId               bigint                  NULL      -- used for main CRUD operation 
 
-    CONSTRAINT PKC_Resource PRIMARY KEY CLUSTERED (ResourceTypeId, ResourceSurrogateId) WITH (DATA_COMPRESSION = PAGE) ON PartitionScheme_ResourceTypeId(ResourceTypeId),
-    CONSTRAINT CH_Resource_RawResource_Length CHECK (RawResource > 0x0)
+    CONSTRAINT PKC_ResourceCurrent_ResourceTypeId_ResourceSurrogateId PRIMARY KEY CLUSTERED (ResourceTypeId, ResourceSurrogateId) WITH (DATA_COMPRESSION = PAGE) ON PartitionScheme_ResourceTypeId (ResourceTypeId)
+   ,CONSTRAINT U_ResourceCurrent_ResourceTypeId_ResourceId UNIQUE (ResourceTypeId, ResourceId) ON PartitionScheme_ResourceTypeId (ResourceTypeId)
 )
 
-ALTER TABLE dbo.Resource SET ( LOCK_ESCALATION = AUTO )
+ALTER TABLE dbo.ResourceCurrent SET ( LOCK_ESCALATION = AUTO )
 
-CREATE INDEX IX_ResourceTypeId_TransactionId ON dbo.Resource (ResourceTypeId, TransactionId) WHERE TransactionId IS NOT NULL ON PartitionScheme_ResourceTypeId (ResourceTypeId)
-CREATE INDEX IX_ResourceTypeId_HistoryTransactionId ON dbo.Resource (ResourceTypeId, HistoryTransactionId) WHERE HistoryTransactionId IS NOT NULL ON PartitionScheme_ResourceTypeId (ResourceTypeId)
+CREATE UNIQUE INDEX IXU_ResourceTypeId_ResourceSurrogateId_WHERE_IsDeleted_0 ON dbo.ResourceCurrent (ResourceTypeId, ResourceSurrogateId) WHERE IsDeleted = 0 ON PartitionScheme_ResourceTypeId (ResourceTypeId)
+CREATE INDEX IX_ResourceTypeId_TransactionId_WHERE_TransactionId_NOT_NULL ON dbo.ResourceCurrent (ResourceTypeId, TransactionId) WHERE TransactionId IS NOT NULL ON PartitionScheme_ResourceTypeId (ResourceTypeId)
+GO
 
-CREATE UNIQUE NONCLUSTERED INDEX IX_Resource_ResourceTypeId_ResourceId_Version ON dbo.Resource
+CREATE TABLE dbo.ResourceHistory
 (
-    ResourceTypeId,
-    ResourceId,
-    Version
-)
-ON PartitionScheme_ResourceTypeId(ResourceTypeId)
+    ResourceTypeId              smallint                NOT NULL
+   ,ResourceSurrogateId         bigint                  NOT NULL
+   ,ResourceId                  varchar(64)             COLLATE Latin1_General_100_CS_AS NOT NULL
+   ,Version                     int                     NOT NULL
+   ,IsHistory                   bit                     NOT NULL CONSTRAINT DF_ResourceHistory_IsHistory DEFAULT 1, CONSTRAINT CH_ResourceHistory_IsHistory CHECK (IsHistory = 1)
```

Comment:
Similar question, aren't all resources in this table historical?

Reply:
Yes. Same answer.